frontend/src/host_orchestrator/orchestrator/instancemanager.go

```diff
@@ -219,9 +222,31 @@ func (m *CVDToolInstanceManager) launchCVD(req apiv1.CreateCVDRequest, op apiv1.
 	result = OperationResult{Value: cvd}
 }
 
+const (
+	defaultBranch = "aosp-master"
+	defaultTarget = "aosp_cf_x86_64_phone-userdebug"
+)
+
 func (m *CVDToolInstanceManager) launchFromAndroidCI(
 	req apiv1.CreateCVDRequest, op apiv1.Operation) (*apiv1.CVD, error) {
-	artifactsDir, err := m.fetchCVDHandler.Fetch(req.CVD.BuildSource.AndroidCIBuild)
+	build := req.CVD.BuildSource.AndroidCIBuild
+	branch := build.Branch
+	if branch == "" {
+		branch = defaultBranch
+	}
+	target := build.Target
+	if target == "" {
+		target = defaultTarget
+	}
+	buildID := build.BuildID
+	if buildID == "" {
+		var err error
+		buildID, err = m.buildAPI.GetLatestGreenBuildID(branch, target)
```

Comment:
Isn't `fetch_cvd` capable of figuring out the last green build from a branch? If that's the case the buildAPI wouldn't be needed at all, right?.

Reply:
With this in mind, it should probably be `fetchCVDHandler` the one to extract the build id from the request, at least from the `BuildSource` property (see comment on line 232)

Reply:
fetch_cvd could do that, however fetchCVDHandler has a a logic in-place based on key `build_id/target` to do not download the same artifacts twice, to continue taking advantage of this logic I created this PR, in which we could get the last green `build id` the same way fetch_cvd does beforehand.

Reply:
can you move the "get last id" logic to fetchCVDHandler then?

Reply:
the build id value is needed in this logic as well.

Reply:
can fetchCVDHandler return it?

Reply:
I prefer to keep the fetchCVDHandler api as minimal as possible where the passed parameters "build_id/target" correspond to a concrete target leaving the responsibility to determine the `build_id` if none was set to the instance manager itself.

Reply:
I agree that a wrapper around `fetch_cvd` (i.e `fetchCVDHandler`) should provide the functionality of `fetch_cvd` and nothing more. However I disagree that it's the responsibility of the `InstanceManager` to interact with one particular build API. I think we're missing a layer between those two. That's something we can brainstorm on later since it's beyond the context of this PR.

Reply:
agree.